src/cfnlint/rules/functions/FindInMap.py

```diff
@@ -59,6 +59,62 @@ def check_dict(self, obj, tree):
 
         return matches
 
+    def map_name(self, map_name, mappings, tree):
+        """ Check the map name """
+        matches = []
+        if isinstance(map_name, (six.string_types, dict)):
+            if isinstance(map_name, dict):
+                matches.extend(self.check_dict(map_name, tree[:] + [0]))
+            else:
+                if map_name not in mappings:
+                    message = 'Map Name {0} doesnt exist for {0}'
+                    matches.append(RuleMatch(
+                        tree[:] + [0], message.format(map_name, '/'.join(map(str, tree)))))
+        else:
+            message = 'Map Name should be a {0}, or string at {1}'
+            matches.append(RuleMatch(
+                tree[:] + [0],
+                message.format(
+                    ', '.join(map(str, self.supported_functions)),
+                    '/'.join(map(str, tree)))))
+
+        return matches
+
+    def first_key(self, first_key, tree):
+        """ Check the validity of the first key """
+        matches = []
+        if isinstance(first_key, (six.string_types, int)):
+            return matches
+        if isinstance(first_key, (dict)):
+            matches.extend(self.check_dict(first_key, tree[:] + [1]))
+        else:
+            message = 'FindInMap first key should be a {0}, string, or int at {1}'
+            matches.append(RuleMatch(
+                tree[:] + [1],
+                message.format(
+                    ', '.join(map(str, self.supported_functions)),
+                    '/'.join(map(str, tree)))))
+
+        return matches
+
+    def second_key(self, second_key, tree):
```

Comment:
Not a huge deal since CFN only supports two levels, but we could make these identical functions a unified “map_key()” func — we’d lose the “first/second” in the message text though, and I seriously doubt CFN will ever add more levels to the map

Reply:
I was thinking about it.  If I combine it into one function I would just pass in an array of the keys.